test/model-method.js

```diff
@@ -101,3 +101,220 @@ describe('ModelMethod', function() {
+            done();
+          });
+        });
+      });
+  });
+
+  it('correctly adds prototype method', function(done) {
+    ModelMethod.create(
+      {
+        modelId: userModel.id,
+        name: 'testMethod',
+        isStatic: false
+      },
+      function(err) {
+        if (err) return done(err);
+        userModel.methods(function(err, list) {
+          if (err) return done(err);
+          expect(list).to.have.length(1);
+          expect(list[0]).to.have.property('name', 'testMethod');
+          expect(list[0]).to.have.property('isStatic', false);
+
+          var cfg = new ConfigFile({ path: 'server/models/user.json' });
+          cfg.load(function(err) {
+            if (err) return done(err);
+            var methods = cfg.data.methods;
+            expect(methods).to.be.an('object');
+            expect(methods).to.have.property('testMethod');
+            expect(methods).to.not.have.property('prototype.testMethod');
+            expect(methods.testMethod).to.have.property('isStatic', false);
+            done();
+          });
+        });
+      });
+  });
+});
+
+describe('ModelMethod - Loopback 3.0', function() {
+  var userModel;
+
+  givenLB3Workspace = function(cb) {
+    givenWorkspaceWithCustomDependencies('empty-server', {
+      loopback: '^3.0.0-alpha.1' 
+    }, cb);
+  };
+
+  beforeEach(function(done) {
+    givenLB3Workspace(done);
+  });
+  
+  beforeEach(function(done) {
+    ModelDefinition.create(
+      {
+        name: 'user',
+        facetName: 'server'
+      },
+      function(err, result) {
+        if (err) return done(err);
+        userModel = result;
+        done();
+      });
+  });
+ 
+  it('add static method without isStatic flag to model definition', function(done) {
+    ModelMethod.create(
+      {
+        modelId: userModel.id,
+        name: 'testMethod',
+        isStatic: true
+      },
+      function(err) {
+        if (err) return done(err);
+        userModel.methods(function(err, list) {
+          if (err) return done(err);
+          expect(list).to.have.length(1);
+          expect(list[0]).to.have.property('name', 'testMethod');
+          expect(list[0]).to.not.have.property('isStatic');
+
+          var cfg = new ConfigFile({ path: 'server/models/user.json' });
+          cfg.load(function(err) {
+            if (err) return done(err);
+            var methods = cfg.data.methods;
+            expect(methods).to.be.an('object');
+            expect(methods).to.have.property('testMethod');
+            expect(methods).to.not.have.property('prototype.testMethod');
+            expect(methods.testMethod).to.not.have.property('isStatic');
+            done();
+          });
+        });
+      });
+  });
+  
+  it('add prototype method without isStatic flag to model definition', function(done) {
+    ModelMethod.create(
+      {
+        modelId: userModel.id,
+        name: 'prototype.testMethod',
+        isStatic: false
```

Comment:
The test name says "without isStatic flag", but the test data does set the flag. PTAL